### PR TITLE
Bda new method bookings

### PR DESCRIPTION
**Pull request by benoitdassy**
Method new for the booking + edit the show view of dragon + add some alerts when we create or delete something + update dragon details on booking

### CODE REVIEW

File: app/controllers/bookings_controller.rb [hunk 1]
```diff
@@ -8,13 +8,40 @@ def index
   def show
   end
 
+  def new
+    @dragon = Dragon.find(params[:dragon_id])
```
Comment (ngrognet) [inline app/controllers/bookings_controller.rb:12]: I suggest creating a 'set_dragon' private method, just like for the set_booking

File: app/controllers/bookings_controller.rb [hunk 2]
```diff
@@ -8,13 +8,40 @@ def index
...
+    @booking = Booking.new
+  end
+
+  def create
+    @dragon = Dragon.find(params[:dragon_id])
```
Comment (ngrognet) [inline app/controllers/bookings_controller.rb:17]: same here

File: app/controllers/bookings_controller.rb [hunk 3]
```diff
@@ -8,13 +8,40 @@ def index
...
+    @booking = Booking.new(booking_params)
+    @booking.dragon = @dragon
+    @booking.user = current_user
+    @booking.status = "Pending"
+    duration = ((DateTime.parse(booking_params[:end_date]).to_date - DateTime.parse(booking_params[:start_date]).to_date).to_i) + 1
+    @booking.duration = duration
```
Comment (neisankun) [inline app/controllers/bookings_controller.rb:23]: this two lines can be merged

File: app/controllers/bookings_controller.rb [hunk 4]
```diff
@@ -8,13 +8,40 @@ def index
...
+      flash[:notice] = "Booking successfully created"
+      redirect_to booking_path(@booking)
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
   def destroy
     @booking.destroy
+    flash[:notice] = "Booking #00#{@booking.id} successful canceled"
```
Comment (ngrognet) [inline app/controllers/bookings_controller.rb:35]: use the rjust ruby method here for padding

Comment (ngrognet) [inline app/controllers/bookings_controller.rb:35]: should be 'successfully canceled'

**Pull request merged**